staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -906,3 +908,107 @@ func TestDispatchEventWillNotBeBlockedByTimedOutWatcher(t *testing.T) {
 		t.Errorf("watcher is blocked by slower one (count: %d)", eventsCount)
 	}
 }
+
+func verifyEvents(t *testing.T, w watch.Interface, events []watch.Event) {
+	_, _, line, _ := goruntime.Caller(1)
+	for _, expectedEvent := range events {
+		select {
+		case event := <-w.ResultChan():
+			if e, a := expectedEvent.Type, event.Type; e != a {
+				t.Logf("(called from line %d)", line)
+				t.Errorf("Expected: %s, got: %s", e, a)
+			}
+			object := event.Object
+			if co, ok := object.(runtime.CacheableObject); ok {
+				object = co.GetObject()
+			}
+			if e, a := expectedEvent.Object, object; !apiequality.Semantic.DeepEqual(e, a) {
+				t.Logf("(called from line %d)", line)
+				t.Errorf("Expected: %#v, got: %#v", e, a)
+			}
+		case <-time.After(wait.ForeverTestTimeout):
+			t.Logf("(called from line %d)", line)
+			t.Errorf("Timed out waiting for an event")
+		}
+	}
+}
+
+func TestCachingDeleteEvents(t *testing.T) {
+	backingStorage := &dummyStorage{}
+	cacher, _, err := newTestCacher(backingStorage, 1000)
+	if err != nil {
+		t.Fatalf("Couldn't create cacher: %v", err)
+	}
+	defer cacher.Stop()
+
+	// Wait until cacher is initialized.
+	cacher.ready.wait()
+
+	fooPredicate := storage.SelectionPredicate{
+		Label: labels.SelectorFromSet(map[string]string{"foo": "true"}),
+		Field: fields.Everything(),
+	}
+	barPredicate := storage.SelectionPredicate{
+		Label: labels.SelectorFromSet(map[string]string{"bar": "true"}),
+		Field: fields.Everything(),
+	}
+
+	createWatch := func(pred storage.SelectionPredicate) watch.Interface {
+		w, err := cacher.Watch(context.TODO(), "pods/ns", "999", pred)
+		if err != nil {
+			t.Fatalf("Failed to create watch: %v", err)
+		}
+		return w
+	}
+
+	w1 := createWatch(storage.Everything)
```

Comment:
nit: name the watchers something like `everythingWatcher`, `fooWatcher`, `barWatcher` to make the later use easier to read

Reply:
done